ui/inspectors/inspector.reel/inspector-confirmation.reel/inspector-confirmation.js

```diff
@@ -8,9 +8,47 @@ var Component = require("montage/ui/component").Component;
  * @extends Component
  */
 exports.InspectorConfirmation = Component.specialize(/** @lends InspectorConfirmation# */ {
-    constructor: {
-        value: function InspectorConfirmation() {
-            this.super();
+
+    handleConfirmDeleteAction: {
+        value: function (event) {
+            this.parentComponent.isConfirmationVisible = false;
+            this.confirmDelete();
+        }
+    },
+
+    handleCancelDeleteAction: {
+        value: function() {
+            this.parentComponent.isConfirmationVisible = false;
+        }
+    },
+
+    confirmDelete: {
+        value: function (event) {
+            var self = this,
+                promise;
+
+            this.parentComponent._isToBeDeleted = true;
+
+            if (typeof this.parentComponent.parentComponent.delete === 'function') {
```

Comment:
Yeah. I originally tried to avoid that (see before this commit). However, I don't know how to get the correct function to call for `handleConfirmDeleteAction` in a way that will avoid even worse scope issues. The consequence of the version before this commit was that if an inspector implemented its own `delete`, it didn't work.